api/server/sdk/schedule_policy_test.go

```diff
@@ -143,6 +143,115 @@ func TestSdkSchedulePolicyCreateSuccess(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestSdkSchedulePolicyCreateSuccessWithDefaultValues(t *testing.T) {
+
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+
+	seconds := int64(1234567)
+	req := &api.SdkSchedulePolicyCreateRequest{
+		SchedulePolicy: &api.SdkSchedulePolicy{
+			Name: "dummy-schedule-name",
+			Schedules: []*api.SdkSchedulePolicyInterval{
+				&api.SdkSchedulePolicyInterval{
+					PeriodType: &api.SdkSchedulePolicyInterval_Daily{
+						Daily: &api.SdkSchedulePolicyIntervalDaily{
+							Hour:   0,
+							Minute: 30,
+						},
+					},
+				},
+				&api.SdkSchedulePolicyInterval{
+					PeriodType: &api.SdkSchedulePolicyInterval_Monthly{
+						Monthly: &api.SdkSchedulePolicyIntervalMonthly{
+							Day:    20,
+							Hour:   11,
+							Minute: 22,
+						},
+					},
+				},
+				&api.SdkSchedulePolicyInterval{
+					PeriodType: &api.SdkSchedulePolicyInterval_Periodic{
+						Periodic: &api.SdkSchedulePolicyIntervalPeriodic{
+							Seconds: seconds,
+						},
+					},
+				},
+				&api.SdkSchedulePolicyInterval{
+					PeriodType: &api.SdkSchedulePolicyInterval_Weekly{
+						Weekly: &api.SdkSchedulePolicyIntervalWeekly{
+							Day:    api.SdkTimeWeekday_SdkTimeWeekdayTuesday,
+							Hour:   10,
+							Minute: 10,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	s.MockCluster().
+		EXPECT().
+		SchedPolicyCreate(req.GetSchedulePolicy().GetName(), gomock.Any()).
+		Do(func(name, schedule string) {
+			// Verify that the yaml string created can be parsed
+			intervals, _, err := sched.ParseScheduleAndPolicies(schedule)
+			assert.NoError(t, err)
+			assert.Len(t, intervals, len(req.GetSchedulePolicy().GetSchedules()))
+
+			// Verify name is correct
+			assert.Equal(t, req.GetSchedulePolicy().GetName(), name)
+
+			// Verify data is correct
+			policies, err := retainInternalSpecYamlByteToSdkSched([]byte(schedule))
+			assert.NoError(t, err)
+			assert.Len(t, policies, len(req.GetSchedulePolicy().GetSchedules()))
+
+			// Check Daily data
+			actualDaily := policies[0]
+
+			fmt.Println(actualDaily.GetRetain())
+			expectedDaily := req.GetSchedulePolicy().GetSchedules()[0]
+			fmt.Println(expectedDaily.GetRetain())
+
+			assert.Equal(t, expectedDaily.GetRetain(), actualDaily.GetRetain())
```

Comment:
So I guess when the SDK calls schedpolicy it does it with retain of 0 still, right?